src/main/subgridMod.F90

```diff
@@ -146,11 +148,19 @@ subroutine subgrid_get_info_natveg(gi, npatches, ncols, nlunits)
 
     npatches = 0
 
-    do pft = natpft_lb, natpft_ub
-       if (natveg_patch_exists(gi, pft)) then
-          npatches = npatches + 1
-       end if
-    end do
+    if(use_fates)then
```

Comment:
maxveg won't be the same as natpft_ub for non-SP runs. Even if we wanted to, maxveg can't be the same as natpft_ub when we have FATES non-SP with no-competition and fixed biogeography.  In this scenario, similar to what I mention above, we have an array of pft weights (dimensioned by natpft_ub) that is the size of the pfts inthe surface file, but maxveg may be different because it represents the actual number of fates patches.

Reply:
OK, we discussed this and actually we realize that we do need to have natpft_ub to give the value from the file. So we'll leave it as it is.

What we did realize we could do is to remove some of the "if(use_fates)" if we have some of these loops go over 0 to maxveg rather than natpft_lb to natpft_ub. The end goal that I wanted was to remove some of the FATES specific logic. It'll b e much better to not have to have logic specific to FATES in the code as much as possible.

Reply:
We also wonder if maxveg should have it's name changed to indicate it's talking about the number of patches rather than the number of PFT's. But, we should leave it in place for now.

We should add a minveg that's set to zero, though so that we don't have hardcoded 0's in the code.